src/sysdir.c

```diff
@@ -85,8 +85,12 @@ static int git_sysdir_guess_template_dirs(git_buf *out)
 
 typedef int (*git_sysdir_guess_cb)(git_buf *out);
 
-static git_buf git_sysdir__dirs[GIT_SYSDIR__MAX] =
-	{ GIT_BUF_INIT, GIT_BUF_INIT, GIT_BUF_INIT, GIT_BUF_INIT, GIT_BUF_INIT };
+struct git_sysdir__dir {
+	int initialized;
+	git_buf buf;
+};
+
+static struct git_sysdir__dir git_sysdir__dirs[GIT_SYSDIR__MAX];
```

Comment:
Hum. If you're already creating a new sysdir struct, what do you think about including the guess callbacks into the struct and then drop the `git_sysdir__dir_guess` array? So basically

```
static struct git_sysdir__dir dirs[] = {
    { git_sysdir_guess_system_dirs, GIT_BUF_INIT },
    { git_sysdir_guess_global_dirs, GIT_BUF_INIT },
    ...
};
```

This would make the correlation of these two structures clearer and reduce a bit of its magic.